### ISSUE TITLE
Borgbackup 1.1.14 -> 1.1.15 + dependency graph question

### ISSUE THREAD

**Issue by kubedzero** [dmacias72/unRAID-NerdPack#38]
Just a heads up that Borg 1.1.15 is released. I've compiled it and gotten it working on Unraid, but it seems that it needs a new library `xxHash-0.8.0` to be installed before it will function properly on Unraid. I've created an [issue](https://github.com/borgbackup/borg/issues/5591) with Borg to see what they think about this or if there's a way to use the built in hash library from previous versions.

I'll hold off on making a PR to update Nerd Pack to the newest version until that issue gets sorted out, but I have it compiled and ready to go so that's one less thing for you to do. 

In the meantime, is there any support in Nerd Pack for having one package installation/enabling force-enable a dependency? For example, if this dependency change for Borg stays the same, would there be a way to force the Borg package installation in Nerd Pack to also install the xxHash package? If not, users will either have to know ahead of time that both are needed, or they'll hit the stacktrace and then figure out that they need this other xxHash package. As another example, llfuse-1.3.8 is another dependency needed for Borg to function.

**Comment by kubedzero** [comment]
The Borg folks got back and referenced  https://github.com/borgbackup/borg/blob/1.1.15/setup.py#L26. If the system compiling Borg has xxHash installed and this flag is set to `True`, it will use the xxHash library. Otherwise, it will use the library bundled with the Borg source code. It does the same thing for libb2 vs blake2, libzstd vs zstd, and liblz4 vs lz4. I'm guessing the non-bundled packages offer architecture-specific performance improvements over the bundled versions. 

Since I was compiling on Slackware-current and not Unraid, some of these libraries such as `xxHash` were installed by default, and thus referenced during compilation. However, since Unraid is pretty stripped down, it didn't have these libraries, and thus ran into the `ImportError: libxxhash.so.0: cannot open shared object file: No such file or directory` issues I saw.

To keep the dependency graph small so Unraid/Nerd Pack users don't need to be bothered by installing extra packages, I've updated the SlackBuild to modify the `setup.py` file to disable checking the system libraries. That way, we have minimal reliance on the state of the OS and instead depend as much as possible on the Borg source code when compiling and running. It may not be able to use every optimization out there, but I figure compatibility trumps optimization in this case. 

With the recompiled version, I've confirmed `xxHash` no longer needs independent installation for Borg to function. I've tested Borg mounting old archives and system-to-system backups, and all looks good. After a bit more testing I'll put out a PR.

**Comment by dmacias72** [comment]
Thanks for doing this. I haven't had much time lately. The llfuse dependency will fix itself once unraid 6.9 is final. The problem is the version 6.8 uses is older. I'm not sure if you saw my build source for BorgBackup in this repo. That's what I use, plus the extra binaries. I compile it in a VM of unraid I spin up with NerdPack and DevPack installed. DevPack has a bunch of dev tools and the missing source for the packages already in unraid.

**Comment by kubedzero** [comment]
> The llfuse dependency will fix itself once unraid 6.9 is final.

Oh, nice! Once that's confirmed, I guess `llfuse-1.3.8` can be removed from the 6.9 Nerd Pack.

> I'm not sure if you saw my build source for BorgBackup in this repo

That's exactly what I used, thanks so much for providing it! Regarding the extra binaries, I had removed them since on Slackware-current that were installed by default, but I can put them back before I send a PR. Regarding DevPack, I had never seen/used it before. I'll have to look into that.
